x-pack/test/api_integration/apis/asset_manager/tests/assets_hosts.ts

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Client } from '@elastic/elasticsearch';
+import {
+  AssetDocument,
+  ESDocumentWithOperation,
+  HostMetricsDocument,
+  infra,
+  PodMetricsDocument,
+  Serializable,
+  timerange,
+} from '@kbn/apm-synthtrace-client';
+import { getDedotTransform } from '@kbn/apm-synthtrace/src/lib/apm/client/apm_synthtrace_es_client/get_dedot_transform';
+import { getSerializeTransform } from '@kbn/apm-synthtrace/src/lib/apm/client/apm_synthtrace_es_client/get_serialize_transform';
+import { Asset, AssetWithoutTimestamp } from '@kbn/assetManager-plugin/common/types_api';
+import expect from '@kbn/expect';
+import { pipeline, Readable, Transform } from 'stream';
+import { ASSETS_ENDPOINT } from '../constants';
+import { FtrProviderContext } from '../types';
+
+const HOST_ASSETS_ENDPOINT = `${ASSETS_ENDPOINT}/hosts`;
+
+type InfraDocuments =
+  | HostMetricsDocument
+  | PodMetricsDocument
+  | AssetDocument<'host'>
+  | AssetDocument<'pod'>;
+
+export default function ({ getService }: FtrProviderContext) {
+  const supertest = getService('supertest');
+  const es = getService('es');
+  const synthtraceEsClient = getService('synthtraceEsClient');
+
+  describe('asset management', () => {
+    describe('GET /assets/hosts', () => {
+      before(async () => {
+        synthtraceEsClient.pipeline(getInfraPipeline());
```

Comment:
is this an async call that we need to await/return ?

Reply:
No, it's just a plain setter. (I've reworked this in the new PR)